File: core/src/main/java/de/d3adspace/mercantor/core/thread/MercantorThreadFactory.java
package de.d3adspace.mercantor.core.thread;

import de.d3adspace.mercantor.core.MercantorConstants;

import java.util.concurrent.ThreadFactory;
import java.util.concurrent.atomic.AtomicInteger;

/**
 * Thread factory used to name the threads properly.
 *
 * @author Felix Klauke <fklauke@itemis.de>
 */
public class MercantorThreadFactory implements ThreadFactory {

    private final AtomicInteger threadIdCounter;

    public MercantorThreadFactory() {
        this(new AtomicInteger());
    }

    public MercantorThreadFactory(AtomicInteger threadIdCounter) {
        this.threadIdCounter = threadIdCounter;
    }

    @Override
    public Thread newThread(Runnable r) {
        Thread thread = new Thread(r);
        thread.setName(MercantorConstants.MERCANTOR_WORKER_THREAD_PREFIX + threadIdCounter.getAndIncrement());
        return thread;
    }
}

File: commons/src/main/kotlin/de/d3adspace/mercantor/commons/thread/MercantorThreadFactory.kt
package de.d3adspace.mercantor.commons.thread

import java.util.concurrent.ThreadFactory
import java.util.concurrent.atomic.AtomicInteger

/**
 * @author Felix Klauke <info@felix-klauke.de>
 */
class MercantorThreadFactory(private val namePrefix: String) : ThreadFactory {

    private val threadIndex = AtomicInteger()

    override fun newThread(runnable: Runnable): Thread {
        val thread = Thread(runnable)
        thread.name = "$namePrefix #${threadIndex.getAndIncrement()}"
        return thread
    }
}
